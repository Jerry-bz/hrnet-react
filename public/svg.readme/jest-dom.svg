<svg xmlns="http://www.w3.org/2000/svg" width="103.61" height="35" viewBox="0 0 103.61 35"><rect class="svg__rect" x="0" y="0" width="103.61" height="35" fill="#2F6209"/><rect class="svg__rect" x="103.61" y="0" width="0" height="35" fill="#8FC965"/><path class="svg__text" d="M13.61 19.57L13.61 19.57L15.09 19.57Q15.09 20.25 15.43 20.59Q15.76 20.93 16.40 20.93L16.40 20.93Q17.00 20.93 17.35 20.54Q17.71 20.14 17.71 19.45L17.71 19.45L17.71 13.47L19.19 13.47L19.19 19.45Q19.19 20.68 18.42 21.40Q17.66 22.12 16.40 22.12L16.40 22.12Q15.07 22.12 14.34 21.44Q13.61 20.77 13.61 19.57ZM29.45 22L23.87 22L23.87 13.47L29.40 13.47L29.40 14.66L25.35 14.66L25.35 17.02L28.85 17.02L28.85 18.19L25.35 18.19L25.35 20.82L29.45 20.82L29.45 22ZM33.21 19.42L33.21 19.42L34.69 19.42Q34.69 20.15 35.17 20.55Q35.65 20.95 36.55 20.95L36.55 20.95Q37.32 20.95 37.71 20.63Q38.10 20.32 38.10 19.80L38.10 19.80Q38.10 19.24 37.70 18.94Q37.31 18.63 36.28 18.32Q35.25 18.01 34.64 17.63L34.64 17.63Q33.47 16.90 33.47 15.72L33.47 15.72Q33.47 14.69 34.31 14.02Q35.15 13.35 36.49 13.35L36.49 13.35Q37.39 13.35 38.08 13.68Q38.78 14.01 39.18 14.61Q39.58 15.22 39.58 15.96L39.58 15.96L38.10 15.96Q38.10 15.29 37.68 14.91Q37.26 14.54 36.48 14.54L36.48 14.54Q35.76 14.54 35.35 14.85Q34.95 15.16 34.95 15.71L34.95 15.71Q34.95 16.18 35.39 16.50Q35.82 16.81 36.82 17.10Q37.81 17.40 38.42 17.78Q39.02 18.16 39.30 18.65Q39.58 19.13 39.58 19.79L39.58 19.79Q39.58 20.86 38.77 21.49Q37.95 22.12 36.55 22.12L36.55 22.12Q35.62 22.12 34.85 21.77Q34.07 21.43 33.64 20.83Q33.21 20.22 33.21 19.42ZM45.55 14.66L42.92 14.66L42.92 13.47L49.68 13.47L49.68 14.66L47.02 14.66L47.02 22L45.55 22L45.55 14.66ZM56.06 18.95L52.99 18.95L52.99 17.80L56.06 17.80L56.06 18.95ZM62.82 22L60.37 22L60.37 13.47L62.89 13.47Q64.02 13.47 64.89 13.97Q65.77 14.48 66.25 15.40Q66.73 16.33 66.73 17.52L66.73 17.52L66.73 17.95Q66.73 19.16 66.25 20.08Q65.76 21.00 64.87 21.50Q63.98 22 62.82 22L62.82 22ZM61.85 14.66L61.85 20.82L62.81 20.82Q63.98 20.82 64.60 20.09Q65.23 19.36 65.24 17.99L65.24 17.99L65.24 17.52Q65.24 16.13 64.64 15.40Q64.03 14.66 62.89 14.66L62.89 14.66L61.85 14.66ZM70.92 18.00L70.92 18.00L70.92 17.52Q70.92 16.28 71.37 15.32Q71.81 14.37 72.61 13.86Q73.42 13.35 74.46 13.35Q75.51 13.35 76.31 13.85Q77.12 14.35 77.56 15.29Q78.00 16.23 78.00 17.48L78.00 17.48L78.00 17.96Q78.00 19.21 77.57 20.16Q77.14 21.10 76.33 21.61Q75.52 22.12 74.47 22.12L74.47 22.12Q73.44 22.12 72.63 21.61Q71.81 21.10 71.37 20.17Q70.93 19.23 70.92 18.00ZM72.41 17.46L72.41 17.96Q72.41 19.36 72.95 20.13Q73.50 20.90 74.47 20.90L74.47 20.90Q75.46 20.90 75.99 20.15Q76.52 19.40 76.52 17.96L76.52 17.96L76.52 17.51Q76.52 16.09 75.98 15.34Q75.45 14.58 74.46 14.58L74.46 14.58Q73.50 14.58 72.96 15.33Q72.42 16.09 72.41 17.46L72.41 17.46ZM83.94 22L82.47 22L82.47 13.47L84.39 13.47L86.85 20.01L89.31 13.47L91.22 13.47L91.22 22L89.74 22L89.74 19.19L89.89 15.43L87.37 22L86.31 22L83.80 15.43L83.94 19.19L83.94 22Z" fill="#FFFFFF"/><path class="svg__text" d="" fill="#FFFFFF" x="116.61"/></svg>