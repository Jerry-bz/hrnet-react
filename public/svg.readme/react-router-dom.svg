<svg xmlns="http://www.w3.org/2000/svg" width="184.7" height="35" viewBox="0 0 184.7 35"><rect class="svg__rect" x="0" y="0" width="184.7" height="35" fill="#2F6209"/><rect class="svg__rect" x="184.7" y="0" width="0" height="35" fill="#8FC965"/><path class="svg__text" d="M15.70 22L14.22 22L14.22 13.47L17.22 13.47Q18.69 13.47 19.50 14.13Q20.30 14.79 20.30 16.05L20.30 16.05Q20.30 16.90 19.89 17.48Q19.47 18.06 18.73 18.37L18.73 18.37L20.65 21.92L20.65 22L19.06 22L17.35 18.71L15.70 18.71L15.70 22ZM15.70 14.66L15.70 17.52L17.22 17.52Q17.97 17.52 18.39 17.15Q18.82 16.77 18.82 16.11L18.82 16.11Q18.82 15.43 18.43 15.05Q18.04 14.68 17.26 14.66L17.26 14.66L15.70 14.66ZM30.27 22L24.69 22L24.69 13.47L30.23 13.47L30.23 14.66L26.18 14.66L26.18 17.02L29.68 17.02L29.68 18.19L26.18 18.19L26.18 20.82L30.27 20.82L30.27 22ZM35.25 22L33.71 22L36.93 13.47L38.26 13.47L41.49 22L39.94 22L39.24 20.01L35.94 20.01L35.25 22ZM37.59 15.28L36.35 18.82L38.83 18.82L37.59 15.28ZM45.17 18.19L45.17 18.19L45.17 17.39Q45.17 16.19 45.60 15.27Q46.03 14.35 46.83 13.85Q47.63 13.35 48.68 13.35L48.68 13.35Q50.09 13.35 50.95 14.12Q51.81 14.89 51.95 16.29L51.95 16.29L50.48 16.29Q50.37 15.37 49.94 14.96Q49.51 14.55 48.68 14.55L48.68 14.55Q47.71 14.55 47.19 15.26Q46.67 15.96 46.66 17.33L46.66 17.33L46.66 18.09Q46.66 19.47 47.16 20.20Q47.65 20.92 48.61 20.92L48.61 20.92Q49.48 20.92 49.92 20.53Q50.36 20.14 50.48 19.22L50.48 19.22L51.95 19.22Q51.82 20.59 50.94 21.35Q50.06 22.12 48.61 22.12L48.61 22.12Q47.59 22.12 46.81 21.63Q46.03 21.15 45.61 20.26Q45.19 19.37 45.17 18.19ZM57.95 14.66L55.31 14.66L55.31 13.47L62.08 13.47L62.08 14.66L59.42 14.66L59.42 22L57.95 22L57.95 14.66ZM68.46 18.95L65.38 18.95L65.38 17.80L68.46 17.80L68.46 18.95ZM74.24 22L72.76 22L72.76 13.47L75.76 13.47Q77.23 13.47 78.04 14.13Q78.84 14.79 78.84 16.05L78.84 16.05Q78.84 16.90 78.43 17.48Q78.01 18.06 77.28 18.37L77.28 18.37L79.19 21.92L79.19 22L77.60 22L75.89 18.71L74.24 18.71L74.24 22ZM74.24 14.66L74.24 17.52L75.76 17.52Q76.51 17.52 76.94 17.15Q77.36 16.77 77.36 16.11L77.36 16.11Q77.36 15.43 76.97 15.05Q76.58 14.68 75.81 14.66L75.81 14.66L74.24 14.66ZM82.97 18.00L82.97 18.00L82.97 17.52Q82.97 16.28 83.41 15.32Q83.85 14.37 84.66 13.86Q85.46 13.35 86.50 13.35Q87.55 13.35 88.35 13.85Q89.16 14.35 89.60 15.29Q90.04 16.23 90.04 17.48L90.04 17.48L90.04 17.96Q90.04 19.21 89.61 20.16Q89.18 21.10 88.37 21.61Q87.56 22.12 86.52 22.12L86.52 22.12Q85.48 22.12 84.67 21.61Q83.86 21.10 83.41 20.17Q82.97 19.23 82.97 18.00ZM84.45 17.46L84.45 17.96Q84.45 19.36 85.00 20.13Q85.54 20.90 86.52 20.90L86.52 20.90Q87.50 20.90 88.03 20.15Q88.56 19.40 88.56 17.96L88.56 17.96L88.56 17.51Q88.56 16.09 88.02 15.34Q87.49 14.58 86.50 14.58L86.50 14.58Q85.54 14.58 85.00 15.33Q84.46 16.09 84.45 17.46L84.45 17.46ZM94.37 19.16L94.37 19.16L94.37 13.47L95.85 13.47L95.85 19.18Q95.85 20.03 96.28 20.48Q96.72 20.93 97.56 20.93L97.56 20.93Q99.27 20.93 99.27 19.13L99.27 19.13L99.27 13.47L100.75 13.47L100.75 19.17Q100.75 20.53 99.88 21.32Q99.01 22.12 97.56 22.12L97.56 22.12Q96.10 22.12 95.23 21.33Q94.37 20.55 94.37 19.16ZM107.00 14.66L104.36 14.66L104.36 13.47L111.13 13.47L111.13 14.66L108.47 14.66L108.47 22L107.00 22L107.00 14.66ZM120.47 22L114.89 22L114.89 13.47L120.42 13.47L120.42 14.66L116.37 14.66L116.37 17.02L119.87 17.02L119.87 18.19L116.37 18.19L116.37 20.82L120.47 20.82L120.47 22ZM126.14 22L124.66 22L124.66 13.47L127.66 13.47Q129.14 13.47 129.94 14.13Q130.74 14.79 130.74 16.05L130.74 16.05Q130.74 16.90 130.33 17.48Q129.92 18.06 129.18 18.37L129.18 18.37L131.09 21.92L131.09 22L129.51 22L127.80 18.71L126.14 18.71L126.14 22ZM126.14 14.66L126.14 17.52L127.67 17.52Q128.42 17.52 128.84 17.15Q129.26 16.77 129.26 16.11L129.26 16.11Q129.26 15.43 128.87 15.05Q128.48 14.68 127.71 14.66L127.71 14.66L126.14 14.66ZM137.76 18.95L134.69 18.95L134.69 17.80L137.76 17.80L137.76 18.95ZM144.52 22L142.06 22L142.06 13.47L144.58 13.47Q145.71 13.47 146.59 13.97Q147.47 14.48 147.95 15.40Q148.43 16.33 148.43 17.52L148.43 17.52L148.43 17.95Q148.43 19.16 147.94 20.08Q147.46 21.00 146.57 21.50Q145.67 22 144.52 22L144.52 22ZM143.55 14.66L143.55 20.82L144.51 20.82Q145.68 20.82 146.30 20.09Q146.93 19.36 146.94 17.99L146.94 17.99L146.94 17.52Q146.94 16.13 146.33 15.40Q145.73 14.66 144.58 14.66L144.58 14.66L143.55 14.66ZM152.62 18.00L152.62 18.00L152.62 17.52Q152.62 16.28 153.06 15.32Q153.51 14.37 154.31 13.86Q155.12 13.35 156.16 13.35Q157.20 13.35 158.01 13.85Q158.81 14.35 159.25 15.29Q159.69 16.23 159.70 17.48L159.70 17.48L159.70 17.96Q159.70 19.21 159.27 20.16Q158.83 21.10 158.03 21.61Q157.22 22.12 156.17 22.12L156.17 22.12Q155.14 22.12 154.32 21.61Q153.51 21.10 153.07 20.17Q152.63 19.23 152.62 18.00ZM154.10 17.46L154.10 17.96Q154.10 19.36 154.65 20.13Q155.20 20.90 156.17 20.90L156.17 20.90Q157.16 20.90 157.69 20.15Q158.22 19.40 158.22 17.96L158.22 17.96L158.22 17.51Q158.22 16.09 157.68 15.34Q157.14 14.58 156.16 14.58L156.16 14.58Q155.20 14.58 154.66 15.33Q154.12 16.09 154.10 17.46L154.10 17.46ZM165.64 22L164.16 22L164.16 13.47L166.09 13.47L168.55 20.01L171.00 13.47L172.92 13.47L172.92 22L171.44 22L171.44 19.19L171.59 15.43L169.07 22L168.01 22L165.49 15.43L165.64 19.19L165.64 22Z" fill="#FFFFFF"/><path class="svg__text" d="" fill="#FFFFFF" x="197.7"/></svg>