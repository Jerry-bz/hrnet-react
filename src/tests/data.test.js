import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import DataEmployees from "../components/data";

// Fake employee
const fakeEmployee = {
  firstName: "John",
  lastName: "Doe",
  startDate: "2020-01-01",
  department: "Sales",
  dateOfBirth: "1990-01-01",
  street: "123 Main St",
  city: "New York",
  zipCode: "10001",
  state: "NY",
};

// Test the DataEmployees component
describe("Checks the display data of employees", () => {
  it("should display employees", () => {
    // Pass the fake employee to the DataEmployees component
    render(
      <BrowserRouter>
        <DataEmployees data={[fakeEmployee]} />
      </BrowserRouter>
    );

    const firstName = screen.getByText(/John/i);
    expect(firstName).toBeInTheDocument();

    const lastName = screen.getByText(/Doe/i);
    expect(lastName).toBeInTheDocument();

    const startDate = screen.getByText(/2020-01-01/i);
    expect(startDate).toBeInTheDocument();

    const department = screen.getByText(/Sales/i);
    expect(department).toBeInTheDocument();

    const dateOfBirth = screen.getByText(/1990-01-01/i);
    expect(dateOfBirth).toBeInTheDocument();

    const street = screen.getByText(/123 Main St/i);
    expect(street).toBeInTheDocument();

    const city = screen.getByText(/New York/i);
    expect(city).toBeInTheDocument();

    const zipCode = screen.getByText(/10001/i);
    expect(zipCode).toBeInTheDocument();

    const state = screen.getByText(/NY/i);
    expect(state).toBeInTheDocument();

    // Compare the snapshot
    expect(DataEmployees).toMatchSnapshot();
  });
});
