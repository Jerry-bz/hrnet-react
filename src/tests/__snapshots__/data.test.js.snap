// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks the display data of employees should display employees 1`] = `[Function]`;
